### DISCUSSION TITLE
Modify ANS

### DISCUSSION THREAD

**Discussion by dionialves** [akvorado/akvorado#1386] [Q&A]
Hello everybody.

First of all, I would like to thank you for the project, it is fantastic.

Secondly, sorry for the English, I'm using Google Translate.

My doubt:
I would like to change the ASN origin number based on the origin IP. I need this, because CDNs from Google, Netflix and others come with the provider's AS and not the AS of the service in question.

**Comment by vincentbernat** [comment]
You would need to build you own GeoIP database in the MMDB format and load it before the regular one. Possibly with [mmdbctl](https://github.com/ipinfo/mmdbctl).

**Reply by dionialves** [reply]
I understood
I will study how to build this database. After building, where in the akvorado would you do this configuration?

**Reply by vincentbernat** [reply]
Here: https://demo.akvorado.net/docs/configuration#geoip

**Reply by dionialves** [reply]
Sorry for insisting on this subject, but I did the configuration and it didn't work. I'll post what I did:

I created a json with the Facebook AS configuration, but in the range I put the CDN IP:
![image](https://github.com/user-attachments/assets/41b5e5a4-4043-48b3-9612-d724e92c83ad)

With the command below I created the database and defined the path /usr/share/GeoIP/asn.mmdb as the destination:
![image](https://github.com/user-attachments/assets/24d33b4c-e6f3-4314-90db-e6c699ace555)

This is the geoip configuration in akvorado:
![image](https://github.com/user-attachments/assets/8e883f67-ecd9-4e49-8c7c-d406642dd2e1)


After configuration, I restarted the composer but the new flows from that IP in question were not associated with the Facebook AS. Could you tell me where my error is?

**Reply by vincentbernat** [reply]
The composer? You need to restart the orchestrator. First, don't modify asn.mmdb. Create your own and put it before the one in configuration:
```yaml
asn-database:
 - /usr/share/GeoIP/custom.mmdb
 - /usr/share/GeoIP/asn.mmdb
```

Then, inside the orchestrator, this is not the same `/usr/share/GeoIP` than on your host. The best way would be to create `docker/docker-compose-local.yml` with:
```yaml
services:
  akvorado-orchestrator:
    volumes:
      - /usr/share/GeoIP/custom.mmdb:/usr/share/GeoIP/custom.mmdb:ro
```
And in .env, add:
```
## Custom settings
COMPOSE_FILE=${COMPOSE_FILE}:docker/docker-compose-local.yml
```

**Comment by dionialves** [comment]
Hello,
I was coming to this conclusion, but I still have difficulties. Here are the settings I've already made:

![image](https://github.com/user-attachments/assets/93dbe0f3-3c47-4479-a892-bc3e1ef721ea)
In this configuration I tried to put the file name, but it still didn't work.

![image](https://github.com/user-attachments/assets/a65161b1-fede-4506-93a0-ec1cc2233274)

![image](https://github.com/user-attachments/assets/63819f9a-c839-4ff6-b81f-6059e52cdff2)

![image](https://github.com/user-attachments/assets/bdb7602d-2341-4f15-866c-5e8183e58f42)

![image](https://github.com/user-attachments/assets/296ffcef-4cbb-4d2a-9e90-f48f2403f90f)
In this image you can see the output of reading the .mmdb file!

I'm studying a lot about it and I can't solve it! I appreciate any help.

**Reply by broonu** [reply]
Did you manage to resolve it? If so, could you share it?

**Comment by GlaucoRibeiro89** [comment]
Check with curl -s http://127.0.0.1:8080/api/v0/orchestrator/configuration/outlet "asnproviders:" stay with:
asnproviders:
  - geo-ip
  - routing
  - flow

the default is:
asnproviders:
  - flow
  - routing
  - geo-ip
